src/util/bip32.rs

```diff
@@ -44,7 +45,8 @@ impl_array_newtype!(Fingerprint, u8, 4);
 impl_bytes_newtype!(Fingerprint, 4);
 
 /// Extended private key
-#[derive(Copy, Clone, PartialEq, Eq, Debug)]
+#[derive(Copy, Clone, PartialEq, Eq)]
+#[cfg_attr(feature = "std", derive(Debug))]
```

Comment:
Why are you doing this? I think this is causing the issue? Why cannot use derive debug in no-std? And how is that related to this PR.

Reply:
We need this since `SecretKey` in the upstream does custom `Debug` which works only on `std`. Without this change it foes not compile on `no_std`